### ISSUE TITLE
Route transition always updates URL lazily

### ISSUE THREAD

**Issue by anobin** [emberjs/ember.js#10917]
I've noticed that when transitioning to a new route the URL will lazily update (ie. not until the route is completely done transitioning). This can create issues with slow or failed transitions as refreshing the page will put you back into the old route.

I did some digging and found that ember has switched between [eager](https://github.com/emberjs/ember.js/pull/4122) and [lazy](https://github.com/emberjs/ember.js/pull/9919) over time.

I also noticed that the docs are currently out-of-date:
"URLs always update immediately unless the transition was aborted or redirected within the same run loop."
http://guides.emberjs.com/v1.11.0/routing/loading-and-error-substates/

I do like the idea of the route *rendering* being eager or lazy based on whether a loading route is provided or not; is there anyway this can be extended to URL updating? Basically, it would be nice if there were some way to configure whether a transition will update the URL eagerly or lazily.

I am currently forcing eager updates by replicating a bit of code from when transitions did update the url eagerly:
```javascript
var transition = this.transitionToRoute('newRoute', {queryParams: queryParams});
                
var newUrl = transition.router.generate('newRoute', {queryParams: transition.queryParams});
if (transition.urlMethod === 'update') {
    transition.router.updateURL(newUrl);
} else if (transition.urlMethod === 'replace') {
    transition.router.replaceURL(newUrl);
}
transition.method(null); // Prevent later update url refire.
```

**Comment by seawatts** [comment]
I'm also running into this issue

**Comment by rwjblue** [comment]
Closing in favor of https://github.com/emberjs/guides/pull/596.
